From d969f7ffaf6d91deb223a27f2b8670f03519979b Mon Sep 17 00:00:00 2001
From: Eric Timmons <etimmons@mit.edu>
Date: Sun, 2 Dec 2018 14:10:23 -0500
Subject: [PATCH 2/3] Fix threads on musl libc

Musl libc does not add any content to _CS_GNU_LIBPTHREAD_VERSION, causing the
runtime check to make sure the thread implementation is NPTL to fail. There
seems to be no way to test not NPTL at run time on musl, so if the configuration
value is empty, just assume the thread implementation is good enough.
---
 src/runtime/linux-os.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/src/runtime/linux-os.c b/src/runtime/linux-os.c
index b9058bdff..6103d79f7 100644
--- a/src/runtime/linux-os.c
+++ b/src/runtime/linux-os.c
@@ -187,8 +187,15 @@ isnptl (void)
       if (strstr (buf, "NPTL")) {
           return 1;
       }
+      else {
+          return 0;
+      }
+  }
+  else {
+      /* If the configuration variable is empty, just assume we have a
+       * good enough thread implementation. */
+      return 1;
   }
-  return 0;
 }
 #endif
 
-- 
2.19.2

